components/layout_2020/positioned.rs

```diff
@@ -482,26 +481,58 @@ impl HoistedAbsolutelyPositionedBox {
                     // https://drafts.csswg.org/css2/visudet.html#abs-replaced-width
                     // https://drafts.csswg.org/css2/visudet.html#abs-replaced-height
                     let style = &replaced.style;
-                    content_size = size.auto_is(|| unreachable!());
+                    content_size = computed_size.auto_is(|| unreachable!());
                     fragments = replaced
                         .contents
                         .make_fragments(style, content_size.clone());
                 },
                 IndependentFormattingContext::NonReplaced(non_replaced) => {
+                    // https://drafts.csswg.org/css2/#min-max-widths
+                    // https://drafts.csswg.org/css2/#min-max-heights
+                    let min_size = non_replaced
+                        .style
+                        .content_min_box_size(&containing_block.into(), &pbm)
+                        .auto_is(|| Length::zero());
+                    let max_size = non_replaced
+                        .style
+                        .content_max_box_size(&containing_block.into(), &pbm);
+
                     // https://drafts.csswg.org/css2/visudet.html#abs-non-replaced-width
                     // https://drafts.csswg.org/css2/visudet.html#abs-non-replaced-height
-                    let inline_size = inline_axis.size.auto_is(|| {
+                    let mut inline_size = inline_axis.size.auto_is(|| {
                         let anchor = match inline_axis.anchor {
                             Anchor::Start(start) => start,
                             Anchor::End(end) => end,
                         };
+                        let margin_sum = inline_axis.margin_start + inline_axis.margin_end;
                         let available_size =
-                            cbis - anchor - pbm.padding_border_sums.inline - margin.inline_sum();
+                            cbis - anchor - pbm.padding_border_sums.inline - margin_sum;
                         non_replaced
                             .inline_content_sizes(layout_context)
                             .shrink_to_fit(available_size)
                     });
 
+                    // If the tentative used inline size is greater than ‘max-inline-size’,
```

Comment:
Might want to link to the specification here since this bit of code is a little tricky.

Reply:
done in 47180bb06493e7a1396be8f048491335d288f4a1